.github/workflows/lint-and-test.yaml

```diff
@@ -3,49 +3,129 @@ name: Lint and test PR
+          done
 
-  docker-build:
+  chart-install:
+    name: Installation test for Helm charts
     runs-on: ubuntu-latest
+    needs:
+      # Reuse image cache:
+      - docker-build
+      # This test is expensive so only run it when cheap tests pass:
+      - chart-lint
+      - chart-unittest
     steps:
-      - uses: actions/checkout@v3
+      - name: Checkout code
+        uses: actions/checkout@v3
+
       - name: Set up Docker Buildx
         uses: docker/setup-buildx-action@v2
-      - name: Build
+      - name: Build image
         uses: docker/build-push-action@v4
         with:
-          tags: ${{ github.repository }}-ci
+          load: true  # Bring image to the host so we can minikube load it later.
+          tags: ci.local/${{ github.repository }}:ci
           cache-from: type=gha
-          cache-to: type=gha,mode=max
+          # No need to push anything to cache here.
+
+      # Install Minikube to test chart installation
+      - name: Install Minikube
+        uses: manusa/actions-setup-minikube@v2.7.2
+        with:
+          minikube version: v1.29.0
+          kubernetes version: v1.26.0
+          github token: ${{ secrets.GITHUB_TOKEN }}
+          driver: docker
+          start args: "--container-runtime=containerd"
+      # Test chart installation with a stable image.
+      - name: Configure testing scenario
+        run: minikube image load ci.local/${{ github.repository }}:ci
+
+      - name: Install helm/chart-testing
+        uses: helm/chart-testing-action@v2.3.1
+      - name: Test charts' installation path
+        run: |
+          ct install --all \
+          --config .github/ct.yaml \
+          --helm-extra-args "--timeout 1m"
+
+    # TODO: Upgrade path.
+    # I need to comment the upghrade path until we have a first release.
+    # Comment for the future me: 
+    #     This also failed because the port was used: set a strategy to recreate.
+    #- name: Test charts' upgrade path
+    #  run: ct --config .github/ct.yaml install --upgrade --all
+
+      # Install charts with the e2e values (tested also in lint-charts job)
+      - name: Configure testing scenario
+        run: |
+          for CHART in charts/*; do
+            helm upgrade --install \
+              "$(basename $CHART)" "$CHART" \
+              --create-namespace --namespace e2e-staless-dns \
+              --values "$CHART/ci/e2e-values.yaml" \
+              --wait --timeout 60s
+          done
+
+      # In this first iteration, readineess probes may not be reliable. Here the CI will wait enough
+      # for powerdns to startup and for external-dns to have to to run (at least) 3 times
+      - name: Wait 40 seconds for everything to settle
+        run: sleep 40
+
+      # Begin e2e test.
+      - name: Basic tests that are not e2e but allow me to test the CI
+        run: |
+          ip=$(minikube ip)
+          
+          kubectl -n e2e-staless-dns get pods || true
+          
+          curl $ip:80 || true
```

Comment:
This won't work trivially, we would need to tell pdns to accept connections from a wider range:
```
Mar 14 18:09:31 [webserver] Webserver closing socket: remote (192.168.49.1) does not match the set ACL(127.0.0.1/32, ::1/128)
```

Reply:
There is a flag `--webserver-allow-from=0.0.0.0/0` missing in the pdns deployment.

Reply:
Done in https://github.com/txqueuelen/stateless-dns/pull/26